src/sequence/mod.rs

```diff
@@ -179,297 +152,140 @@ impl Data {
         end: Index,
         requester: Option<PublicKey>,
     ) -> Result<Option<Entries>> {
-        self.check_permission(Action::Read, requester, None)?;
+        self.check_permission(Action::Read, requester)?;
 
-        Ok(match &self.data {
+        let items = match &self.data {
             SeqData::Public(data) => data.in_range(start, end),
             SeqData::Private(data) => data.in_range(start, end),
-        })
+        };
+
+        let entries = match items {
+            None => None,
+            Some(entries) => Some(entries.iter().map(|(_, entry)| entry.to_vec()).collect()),
```

Comment:
I think we should be exposing the Markers here to the application to avoid race conditions when appending data